src/layouts/Header/Header.jsx

```diff
@@ -1,33 +1,55 @@
+import React from "react";
+
 import clsx from "clsx";
+import { Link, useLocation } from "react-router-dom";
+
+import Logo from "../../assets/icons/logo.svg";
 
 import styles from "./Header.module.css";
 
-const ResponsiveElement = ({
-  children,
-  visibleOn = ["mobile", "tablet", "desktop"],
-}) => {
-  const hiddenOnMobile = !visibleOn.includes("mobile");
-  const hiddenOnTablet = !visibleOn.includes("tablet");
-  const hiddenOnDesktop = !visibleOn.includes("desktop");
+const BREAKPOINTS = {
+  DESKTOP: "desktop",
+  TABLET: "tablet",
+  MOBILE: "mobile",
+};
+
+const ALL_BREAKPOINTS = Object.values(BREAKPOINTS);
+
+const ResponsiveElement = ({ children, visibleOn = ALL_BREAKPOINTS }) => {
+  const hiddenOnDesktop = !visibleOn.includes(BREAKPOINTS.DESKTOP);
+  const hiddenOnTablet = !visibleOn.includes(BREAKPOINTS.TABLET);
+  const hiddenOnMobile = !visibleOn.includes(BREAKPOINTS.MOBILE);
 
   return (
     <div
-      className={clsx({
-        [styles.hidden_on_desktop]: hiddenOnDesktop,
-        [styles.hidden_on_tablet]: hiddenOnTablet,
-        [styles.hidden_on_mobile]: hiddenOnMobile,
+      className={clsx(styles.container, {
+        [styles.hiddenOnDesktop]: hiddenOnDesktop,
+        [styles.hiddenOnTablet]: hiddenOnTablet,
+        [styles.hiddenOnMobile]: hiddenOnMobile,
       })}
     >
       {children}
     </div>
   );
 };
 
-const Header = ({ visibleOn }) => {
+const showButtonPageList = ["/", "/list"];
```

Comment:
라우팅 경로는 상수화를 해서 다뤄주시면 좋아요.
```suggestion
const showButtonPageList = [ROUTE.홈, ROUTE.리스트];
const showButtonPageList = [ROUTE.HOME, ROUTE.LIST];
```